vendors/nuvoton/sdk/middleware/wifi_esp8266/esp8266_wifi.c

```diff
@@ -915,6 +1144,49 @@ ESP_WIFI_Status_t ESP_WIFI_Send( ESP_WIFI_Object_t * pxObj, ESP_WIFI_Conn_t * px
     return xRet;
 }
 
+/**
+ * @brief  Receive Data for TCP passive mode
+ * @retval Operation status
+ */
+ESP_WIFI_Status_t ESP_WIFI_Recv_TCP_PASSIVE( ESP_WIFI_Object_t * pxObj, uint8_t ucLinkID, uint8_t * pcBuf, 
+                                 uint16_t usReqLen, uint16_t * usRecvLen, TickType_t xTickEnd )
+{
+    ESP_WIFI_Status_t xRet = ESP_WIFI_STATUS_TIMEOUT;
+   
+
+    if (xWifiIpd[ucLinkID].DataAvailable != 0) {
+        xWifiIpd[ucLinkID].TcpData = (char *)pcBuf;
+        xWifiIpd[ucLinkID].DataReceive = 0;
+        pxObj->ActiveSocket = ucLinkID;
+
+        // +CIPRECVDATA supports up to 2048 bytes at a time
+        usReqLen = usReqLen > 2048 ? 2048 : usReqLen;
+
+        sprintf((char *)pxObj->CmdData, "AT+CIPRECVDATA=%d,%d\r\n", ucLinkID, usReqLen);
+        xRet = ESP_AT_Command(pxObj, pxObj->CmdData, 0);
+
+        if (xRet == ESP_WIFI_STATUS_OK) {
+            if (xWifiIpd[ucLinkID].DataReceive > 0) {
+                if (xWifiIpd[ucLinkID].DataAvailable > xWifiIpd[ucLinkID].DataReceive) {
+                    xWifiIpd[ucLinkID].DataAvailable -= xWifiIpd[ucLinkID].DataReceive;
+                } else {
+                    xWifiIpd[ucLinkID].DataAvailable = 0;
+                }
+                (*usRecvLen) = xWifiIpd[ucLinkID].DataReceive;
+            }
+        }
+        pxObj->ActiveSocket = -1;
+    } else {
+        // Try to get the +IPD message
+        xRet = ESP_IO_Recv(pxObj, pxObj->CmdData, sizeof(pxObj->CmdData), ucLinkID, ESP_WIFI_SHORT_RECV_TO);
```

Comment:
Hi @sukhmanm ,
 Thanks of your suggestion, I fixed the timeout unit as tick type and also rebase the branch onto master.
 Locally test, it could pass the timeout issue.